src/odemis/gui/cont/tabs.py

```diff
@@ -4087,14 +4090,13 @@ def _onManualFocusReady(self, future):
             istream = self.panel.cmb_focus_detectors.GetSelection()
             self._focus_streams[istream].should_update.value = True
 
-        align_mode = self.tab_data_model.align_mode.value
-        if align_mode == "lens-align":
-            self._enableFocusComponents(manual=True, ccd_stream=False)
-
     def _onManualFocusFinished(self, future):
         """
         Called when finishing manual focus is done
         """
+        if future.cancelled():
```

Comment:
Thanks for catching these erratic cases. 
We talked also about the gauge when cancelled, being stopped in its track. 
Maybe it can be resetted here with:  
`self.panel.gauge_autofocus.SetValue(0)`

Reply:
It sounds good. So I've tried... and it didn't work. I think it's because the ProgressiveFutureConnector also sees the end of the Future, and updates one last time the gauge, just a tiny bit later.
So I've tried to put it into `onAlignMode()`, that works, but then the gauge is *always* reset, even after successfully stopping the manual focus, or completion of the autofocus. That's a little be too often.
So I've tried to put it into Show(True), and that seems to work :-)

Reply:
Nice, it works on switching the tabs, though won't work on switching align mode (Mirror, Centering), but I think it's fine to see it stopped.

Reply:
Indeed, I had forgotten that changing the align mode would also cancel the focus. I guess if we really wanted to fix this case, one way would be to adjust the ProgressiveFutureConnector, with a special parameter indicating that when the future is cancelled we want the gauge to be set to 0 (instead of leaving it as-is).  For now, let's just leave it as-is.